submission/pipelines/run_pipeline.py

```diff
@@ -114,38 +117,45 @@ def run_pipeline_run(
         '',
         '--pipelines-path', pipelines_dir,
         'runtime',
-        '--datasets', DATASETS_DIR,
+        '--datasets', config.DATASETS_DIR,
         '--context', 'TESTING',
-        '--worker-id', WORKER_ID,
+        '--worker-id', config.WORKER_ID,
         'fit-score',
         '--input-run', pipeline_run_path,
         '--output-run', pipeline_rerun_path,
     ]
     if should_output_scores:
         rerun_args += ['--scores', pipeline_rerun_scores_path]
-    cli.main(rerun_args)
+    try:
+        cli.main(rerun_args)
+    except Exception as e:
+        # Let's clean up after ourselves. We don't want to keep documents
+        # for pipeline runs that failed.
+        os.remove(pipeline_run_path)
+        os.remove(pipeline_rerun_path)
+        raise e
 
```

Comment:
does this give us the real pipeline error or just some failure error?

Reply:
`raise e` reraises the original error, so we get the real error message. I'm just inserting a clean-up step in-between.